mir,mir_2_15: Fix builds, modernise & fix VM tests

OPNA2608 opened the pull request:
#### 1st:

`mir` and `mir_2_15` currently fail on master, for different reasons.

- `mir`: Borked test
- `mir_2_15`: Borked test + glog dropped pkg-config generation + new glog version errors out when used: https://github.com/NixOS/nixpkgs/pull/371915#issuecomment-2592669676

Fix these builds, to unblock Mir-based compositors (Miriway & miracle-wm) and the Lomiri DE.

#### 2nd:

Additionally, Alacritty, when launched in the VM tests, complains about the YAML config format:
```
machine # [   24.987840] miriway[1098]: [0.004055543s] [WARN ] [alacritty] YAML config "/etc/xdg/alacritty/alacritty.yml" is deprecated, please migrate to TOML using `alacritty migrate`
```

A `lib.generators.toTOML` was rejected in the past, so switch to `pkgs.formats.yaml`. For consistency, switch the INI file generation to `pkgs.formats.ini` as well.

#### 3rd:

OfBorg fails to run the VM tests. Try some things to hopefully resolve this.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

OPNA2608 committed:
nixosTests.miriway: Update alacritty config format, switch to pkgs.formats

OPNA2608 committed:
nixosTests.miracle-wm: Update alacritty config format, switch to pkgs.formats

OPNA2608 committed:
nixosTests.miriway: Switch back to commented-out code

It seems to work again.

lucasew commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 374873`

OPNA2608 committed:
nixosTests.miriway: Move the mouse to work around VM setup issues

OPNA2608 committed:
nixosTests.miracle-wm: Move the mouse to work around VM setup issues

OPNA2608 commented:
@lucasew `nixpkgs-review` doesn't run VM tests, so that isn't useful here. Please build `nixosTests.miracle-wm` and `nixosTests.miriway` if you want to give a meaningful review of this PR.

lucasew commented:
> @lucasew `nixpkgs-review` doesn't run VM tests, so that isn't useful here. Please build `nixosTests.miracle-wm` and `nixosTests.miriway` if you want to give a meaningful review of this PR.

That's what I did
![Screenshot_20250119-092511](https://github.com/user-attachments/assets/8cc0024f-b452-403b-823d-4a44526cbf85)

OPNA2608 committed:
mir,mir_2_15: Disable UdevWrapperTest.UdevMonitorDoesNotTriggerBeforeEnabling

Failure seems to have been noticed by upstream, I assume they'll look into it:
https://github.com/canonical/mir/issues/3716#issuecomment-2580698552

OPNA2608 commented:
> That's what I did

Then I would've expected your `nixpkgs-review` results to actually state that, like it does for me. Or at least a note stating that you did run them, in case your `nixpkgs-review` - for whatever reason - doesn't state it.

---

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 374873`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 tests built:</summary>
  <ul>
    <li>nixosTests.miracle-wm</li>
    <li>nixosTests.miriway</li>
  </ul>
</details>

---

And while this output *claims* that the tests built successfully, they actually both failed… In a way that looks like something broke somewhere else. :confused:

```
[nix-shell:~/.cache/nixpkgs-review/pr-374873-2]$ ls failed_results/
nixosTests.miracle-wm-x86_64-linux  nixosTests.miriway-x86_64-linux
```
```
machine: making screenshot /nix/store/8pip6l7j10wji50id560r3d7birf3ziw-vm-test-run-miriway/miriway_launched.png
(finished: making screenshot /nix/store/8pip6l7j10wji50id560r3d7birf3ziw-vm-test-run-miriway/miriway_launched.png, in 0.05 seconds)
machine: waiting for (alice|machine) to appear on screen
cleanup
kill machine (pid 9)
qemu-system-x86_64: terminating on signal 15 from pid 6 (/nix/store/qrc496n6fsqp4p5m5h8wmw5d5jwyw5mr-python3-3.12.8/bin/python3.12)
kill vlan (pid 7)
(finished: cleanup, in 0.01 seconds)
Traceback (most recent call last):
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/bin/.nixos-test-driver-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/__init__.py", line 146, in main
    driver.run_tests()
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 174, in run_tests
    self.test_script()
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 166, in test_script
    exec(self.tests, symbols, None)
  File "<string>", line 12, in <module>
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 999, in wait_for_text
    retry(screen_matches, timeout)
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 130, in retry
    if fn(False):
       ^^^^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 988, in screen_matches
    variants = self.get_screen_text_variants()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 963, in get_screen_text_variants
    return self._get_screen_text_variants([0, 1, 2])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 949, in _get_screen_text_variants
    return _perform_ocr_on_screenshot(screenshot_path, model_ids)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 111, in _perform_ocr_on_screenshot
    raise Exception(f"TIFF conversion failed with exit code {ret.returncode}")
Exception: TIFF conversion failed with exit code 1
```

OPNA2608 committed:
mir,mir_2_15: Drop glog dependency

glog pkg-config file disappeared after https://github.com/NixOS/nixpkgs/pull/364543.

- mir doesn't depend on it anymore
- mir_2_15 did, and re-enabling the pkg-config file for the new glog version errored out

Upstream decided to just drop glog as a dependency, so follow suit.

OPNA2608 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 374873`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>lomiri.lomiri</li>
    <li>lomiri.lomiri-session</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lomiri.qtmir</li>
    <li>lomiri.qtmir.dev</li>
    <li>mir</li>
    <li>mir.dev</li>
    <li>mir_2_15</li>
    <li>mir_2_15.dev</li>
    <li>miracle-wm</li>
    <li>miriway</li>
  </ul>
</details>

---

`lomiri.lomiri` has dependencies that are FTBFS on master, `lomiri.lomiri-session` depends on `lomiri.lomiri`. Going ahead with merging this already, since it's blocking running reviews for most Lomiri packages.
